src/hooks/useAsset/useAsset.tsx

```diff
@@ -0,0 +1,109 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+
+import { AllChannelFieldsFragment, AssetAvailability, VideoFieldsFragment } from '@/api/queries'
+import { AssetType, useStorageProviders } from '@/hooks'
+import { createStorageNodeUrl } from '@/utils/asset'
+
+type UseAssetDataArgs =
+  | { entity?: VideoFieldsFragment | null; assetType: AssetType.THUMBNAIL | AssetType.MEDIA }
```

Comment:
We would normally use typescript string literal instead of an enum in case like this. Something like `assetType: 'thumbnail' | 'media'`. Do you think using enum has some advantage?

Reply:
I don't like comparing strings, In may opinion this is antipattern. First of all, you can use enum as a type so you don't have to list all available strings one by one `assetType: AssetType` vs `assetType: 'thumbnail' | 'avatar' | 'media'`. Second, if needed you can always easily change enum instead of finding every string in the code. I know that none of the above applies to this case but in my opinion this is just a good practice.